lib/libutils/ext/include/compiler.h

```diff
@@ -226,4 +226,6 @@
 #define __compiler_atomic_store(p, val) \
 	__atomic_store_n((p), (val), __ATOMIC_RELAXED)
 
+#define READ_ONCE(p) __compiler_atomic_load(p)
```

Comment:
This should go into `core/include/io.h`
`<compiler.h>` just provides the primitives.